### ISSUE TITLE
SVM training failed

### ISSUE THREAD

**Issue by minghaohsu410168** [yzrobot/cloud_annotation_tool#23]
Hello,

Thanks for the nice tool, but I want to ask how to train the svm model, I tried training the model using lcas_simple_data.zip but always got "train failed", and the console got "mv cannot stat pedestrian.* no such file or dictionary".

How should I do, can you give me some advice, thank you.

**Comment by minghaohsu410168** [comment]
@yicheng6o6 
I think you didn't generate the **pedestrian** file. You need to extract the feature manually (enter the Object ID, choose the Object Class and click the Extract Feature button). After feature extraction, an **pedestrain** file will be generated in build/svm folder. Then you can train the svm model.

Yes, the other bounding box without labels are dontcare label. You can check the generated **pedestrian** file, and it will be
<label> <index>:<value> format which comply with LIBSVM (For example 1 1 : 0.5 2 : 0.3). Adding the negative examples maens that you need some **-1** label (For example -1 1 : 0.01 2 : 0.02).

**Comment by yicheng6o6** [comment]
@minghaohsu410168 
Thank you very much for your reply.

Now I can generate the **pedestrian** file and **other classes** file in build/svm folder.

But when I checked these files, I found all classes were labeled as **"1"**. 
![svmp4](https://user-images.githubusercontent.com/92355515/137089380-bd2c07a4-d1f2-4a15-af36-80ab868a2b00.png)

From my understanding, the classes of: **dontcare**, **car** and **wheelchair** should be **"-1"** (non-human), namely I suppose **adding the negative examples** means **annotate the object with non-human classes**.
Is my understanding correct? If not, May i have your suggestions?

Best wishes,
Yicheng

**Comment by minghaohsu410168** [comment]
Yes, if you want to use the model to detect pedestrian only, the **car**, **dontcare** are -1 label. You can modify the feature extraction code in viewer.cpp.
